src/NServiceBus.Hosting.Windows/GenericHost.cs

```diff
@@ -70,27 +71,67 @@ public void Start()
         }
 
         /// <summary>
-        ///     Finalize
+        /// Shut down bus or give up after 30 seconds, so host won't hang indefinitely.
```

Comment:
@dvdstelt did you confirm that the win service hangs forever if Stop never completes?